linfa-svm/src/regression.rs

```diff
@@ -290,17 +292,33 @@ pub mod tests {
             sin_curve[(i, 0)] = *val;
         }
 
-        let kernel = Kernel::params()
-            .method(KernelMethod::Gaussian(50.))
-            .transform(sin_curve.view());
+        let dataset = DatasetBase::new(sin_curve.view(), target.view());
 
-        let dataset = DatasetBase::new(kernel, target.view());
-
-        let model = Svm::params().nu_eps(2., 0.01).fit(&dataset);
+        let model = Svm::params()
+            .nu_eps(2., 0.01)
+            .with_gaussian(50.)
+            .fit(&dataset);
 
         println!("{}", model);
 
-        let predicted = Array1::from(model.predict(sin_curve.clone()));
+        let predicted = Array1::from(model.predict(sin_curve));
         assert!(predicted.mean_squared_error(&target.view()) < 1e-2);
     }
+
+    #[test]
+    fn test_regression_linear_kernel() {
+        // simple 2d straight line
+        let targets = Array::linspace(0f64, 10., 100);
+        let records = targets.clone().into_shape((100, 1)).unwrap();
+
+        let dataset = (records, targets).into();
+
+        // Test the precomputed dot product in the linear kernel case
+        let model = Svm::params().nu_eps(2., 0.01).with_linear().fit(&dataset);
+
+        println!("{}", model);
+
+        let predicted = Array1::from(model.predict(dataset.records()));
+        assert!(predicted.mean_squared_error(&dataset.targets().view()) < 1e-2);
+    }
 }
```

Comment:
This is the test I meant. Without computing the correct values of alpha for regression the mean squared error is around 5000